tests/ut/py/test_runtime_builder.py

```
@@ -615,6 +615,70 @@ def _raise_bad_pin():
+    exit, so the invariant is asserted here rather than left to a crash.
+    """
+
+    def _library(self, path: Path) -> Path:
+        """Compile a trivial shared object at `path`."""
+        import subprocess  # noqa: PLC0415
+
+        source = path.with_suffix(".c")
+        source.write_text("int probe(void) { return 7; }\n", encoding="utf-8")
+        # gcc is a fixed test-only toolchain dependency, not user input.
+        subprocess.run(  # noqa: S603, S607
+            ["gcc", "-shared", "-fPIC", "-o", str(path), str(source)], check=True
```

Ruff (0.16.1)

[error] Starting a process with a partial executable path

(S607)